lib/aryn-sdk/aryn_sdk/partition/partition.py

```diff
@@ -0,0 +1,189 @@
+from typing import BinaryIO, Optional, Union
+from collections.abc import Mapping
+from aryn_sdk.config import ArynConfig
+import requests
+import json
+import logging
+import pandas as pd
+import numpy as np
+from collections import OrderedDict
+
+
+# URL for Aryn Partitioning Service (APS)
+APS_URL = "https://api.aryn.cloud/v1/document/partition"
+
+
+def partition_file(
+    file: BinaryIO,
+    aryn_api_key: Optional[str] = None,
+    aryn_config: ArynConfig = ArynConfig(),
+    threshold: Optional[float] = None,
+    use_ocr: bool = False,
+    extract_table_structure: bool = False,
+    extract_images: bool = False,
+    selected_pages: Optional[list[int]] = None,
```

Comment:
I think this needs to be a more complicated type. Something like: `Optional[list[Union[int, list[int]]]]`. Per Mark, you can pass in page selections like [1, [3,5], 17], which would process pages 1, 3-5 inclusive, and 17.

Reply:
oh, neat